griptape/artifacts/text_artifact.py

```diff
@@ -16,26 +16,28 @@ class TextArtifact(BaseArtifact):
     value: str = field(converter=str, metadata={"serializable": True})
     encoding: str = field(default="utf-8", kw_only=True)
     encoding_error_handler: str = field(default="strict", kw_only=True)
-    _embedding: list[float] = field(factory=list, kw_only=True)
-
-    @property
-    def embedding(self) -> Optional[list[float]]:
-        return None if len(self._embedding) == 0 else self._embedding
+    embedding: Optional[list[float]] = field(default=None, kw_only=True)
 
     def __add__(self, other: BaseArtifact) -> TextArtifact:
         return TextArtifact(self.value + other.value)
 
     def __bool__(self) -> bool:
         return bool(self.value.strip())
 
-    def generate_embedding(self, driver: BaseEmbeddingDriver) -> Optional[list[float]]:
-        self._embedding.clear()
-        self._embedding.extend(driver.embed_string(str(self.value)))
+    def to_text(self) -> str:
+        return self.value
+
+    def to_bytes(self) -> bytes:
+        return str(self.value).encode(encoding=self.encoding, errors=self.encoding_error_handler)
+
+    def generate_embedding(self, driver: BaseEmbeddingDriver) -> list[float]:
+        if self.embedding is None:
+            self.embedding = []
```

Comment:
should we keep it `None` until the embedding driver returns? could be an error

Reply:
Great catch.